bids/modeling/statsmodels.py

```diff
@@ -690,17 +678,55 @@ def _inputs_to_df(self, inputs):
                 input_df.loc[input_df.index[i], con.name] = 1
         return input_df
 
-    def _build_contrasts(self, int_name):
-        """Contrast list of ContrastInfo objects based on current state."""
-        contrasts = {}
+    def _build_contrasts(self, unique_in_contrast=None):
+        """Contrast list of ContrastInfo objects based on current state.
+        
+        Parameters
+        ----------
+        unique_in_contrast : string
+            Name of unique incoming contrast inputs (i.e. if there is only 1)
+        """
+        in_contrasts = self.node.contrasts.copy()
         col_names = set(self.X.columns)
-        for con in self.node.contrasts:
-            name = con["name"]
+
+        # Create dummy contrasts as regular contrasts
+        dummies = self.node.dummy_contrasts
+        if dummies:
+            if 'conditionlist' in dummies:
+                conditions = set(dummies['condition_list'])
+            else:
+                conditions = col_names
+
+            for col_name in conditions:
+                in_contrasts.insert(0, 
+                    {
+                        'name': col_name,
+                        'condition_list': [col_name],
+                        'weights': [1],
+                        'test': dummies.get('test')
+                    }
+                )
+
+        # Process all contrasts, starting with dummy contrasts 
+        # Dummy contrasts are replaced if a contrast is defined with same name
+        contrasts = {}
+        for con in in_contrasts:
             condition_list = list(con["condition_list"])
-            if 1 in condition_list and int_name is not None:
-                condition_list[condition_list.index(1)] = int_name
-            if name == 1 and int_name is not None:
-                name = int_name
+
+            # Rename special 1 construct
+            condition_list = ['intercept' if i == 1 else i for i in condition_list]
```

Comment:
I think this is going to fail:

```YAML
{
  ...
  "DummyContrasts": {"Test": "t"}
}
```

The reason is that instead of changing the `1` in the design matrix to match the column name, it's now going to be `"intercept"`, and when you expand the dummy contrasts to proper contrasts, then the `"name"` will be `"intercept"` so we will never pick the contrast name back up. There's also nothing stopping somebody from explicitly creating a contrast named `"intercept"`, so detecting this case will now be harder.

Reply:
ah, you're right. I am passing the name of the unique variable to this function, so I can rename the "intercept" contrast the name of the unique contrast grouping variable in this special case. i think i meant to do that but forgot. 

however, about catching an explicitly named "intercept" contrast, i have two suggestions:

1) the user can deal w/ the consequences
2) throw an error if any explicitly named contrast conflict w/ dummycontrasts. 

the spec currently says that explicilt contrasts over-ride dummycontrasts, so 2) would be most consistent